pkg/server/util/regentryutil/fetch.go

```diff
@@ -10,90 +10,36 @@ import (
 	"github.com/spiffe/spire/proto/spire/common"
 )
 
-func FetchRegistrationEntries(ctx context.Context, dataStore datastore.DataStore, spiffeID string) ([]*common.RegistrationEntry, error) {
-	fetcher := newRegistrationEntryFetcher(dataStore)
-	return fetcher.Fetch(ctx, spiffeID)
-}
-
-type registrationEntryFetcher struct {
-	dataStore datastore.DataStore
-}
-
-func newRegistrationEntryFetcher(dataStore datastore.DataStore) *registrationEntryFetcher {
-	return &registrationEntryFetcher{
-		dataStore: dataStore,
-	}
-}
-
-func (f *registrationEntryFetcher) Fetch(ctx context.Context, id string) ([]*common.RegistrationEntry, error) {
-	entries, err := f.fetch(ctx, id, make(map[string]bool))
+func FetchRegistrationEntries(ctx context.Context, ds datastore.DataStore, agentID string) ([]*common.RegistrationEntry, error) {
```

Comment:
Maybe it's worth to have a comment for this function describing the process here? That would help to prevent any regression of this improvement when updating the function.
Having comments for `fetchMappedEntries` and `fetchDescendantEntries` would also be helpful.

Reply:
I agree that would be very helpful in understanding what is happening. I'll add one.